pypeit/core/wavecal/identify.py

```diff
@@ -0,0 +1,770 @@
+                        self.linelist_update(widx)
+                        self._slidell.set_val(self._slideval)
+                elif self._end != self._start:
+                    # The mouse button was dragged
+                    if axisID == 0:
+                        if self._start > self._end:
+                            tmp = self._start
+                            self._start = self._end
+                            self._end = tmp
+                        self.update_regions()
+        # Now plot
+        trans = mtransforms.blended_transform_factory(self.ax.transData, self.ax.transAxes)
+        self.canvas.restore_region(self.background)
+        self.draw_fitregions(trans)
+        self.draw_lines()
+        self.canvas.draw()
+
+    def key_press_callback(self, event):
+        """ What to do when a key is pressed
+
+        Parameters
+        ----------
+        event : Event instance
+          matplotlib event instance containing information about the event
+        """
+        # Check that the event is in an axis...
+        if not event.inaxes:
+            return
+        # ... but not the information box!
+        if event.inaxes == self.axi:
+            return
+        axisID = self.get_axisID(event)
+        self.operations(event.key, axisID)
+
+    def operations(self, key, axisID):
+        """ Canvas operations
+
+        Parameters
+        ----------
+        key : str
+          Which key has been pressed
+        axisID : int
+          The index of the axis where the key has been pressed (see get_axisID)
+        """
+        # Check if the user really wants to quit
+        if key == 'q' and self._qconf:
+            if self._changes:
+                self.update_infobox(message="WARNING: There are unsaved changes!!\nPress q again to exit", yesno=False)
+                self._qconf = True
+            else:
+                msgs.bug("Need to change this to kill and return the results to PypeIt")
+                plt.close()
+        elif self._qconf:
+            self.update_infobox(default=True)
+            self._qconf = False
+
+        # Manage responses from questions posed to the user.
+        if self._respreq[0]:
+            if key != "y" and key != "n":
+                return
+            else:
+                # Switch off the required response
+                self._respreq[0] = False
+                # Deal with the response
+                if self._respreq[1] == "write":
+                    # First remove the old file, and save the new one
+                    msgs.work("Not implemented yet!")
+                    self.write()
+                else:
+                    return
+            # Reset the info box
+            self.update_infobox(default=True)
+            return
+
+        if key == '?':
+            print("===============================================================")
+            print("       MAIN OPERATIONS")
+            print("cursor  : Select lines (LMB click)")
```

Comment:
I might suggest you make this a `dict`, put it at the top of the module
and then maybe we can find a way to dynamically make docs for it..

Reply:
I'm about to do the same for the `specobj` data model, so you can assign
it to me.